Release v1.18.0 (#1483)
heremaps/here-data-sdk-cpp: CHANGELOG.md

## v1.18.0 (05/03/2024)
**Common**
* Changed the code to support latest versions of popular compilers.
* Changed the `olp::http::NetworkCurl` to use 4x more opened connection than requests to avoid issues with connections caching.
* Extended `olp::client::RetrySettings` with `connection_timeout` and `transfer_timeout`.
* Added performance improvements.
* Changed minimal required version of `cURL` for SSL blobs support to 7.77.0.
* Added support of background downloads for iOS.
* Fixed possible data-race in `olp::cache::DefaultCache` when the cache size is being checked.
* Added `OLP_SDK_ENABLE_OFFLINE_MODE` CMake option to enforce full offline mode for the SDK.
* Changed the `olp::cache::DiskCache::Open` to return `olp::cache::OpenResult::Postponed` when the protected cache could not be initialized right now due to restrictions like missing permissions.
* Changed the `olp::cache::DefaultCache::Open` to scan cache path for the unexpected directories and report them as errors.

**olp-cpp-sdk-authentication**

> * Now, the credentials file is being correctly parsed when has unexpected line endings (e.g. when file had been copied from one system to another one).

Reviewer: ```suggestion
* The credentials file is parsed correctly when it has unexpected line endings which can occur when copying the file between different operating systems.
```

**olp-cpp-sdk-dataservice-read**

> * Fixed possible crash in `olp::dataservice::read::PartitionsCacheRepository` when the stream was closed before parsing had been started.

Reviewer: ```suggestion
* Fixed possible crash in `olp::dataservice::read::PartitionsCacheRepository` that occurred when the stream closed before parsing started.
```

* Changed the `olp::dataservice::read::repository::PartitionsSaxHandler` to be cancellable.
* Extended `olp::dataservice::read::VersionedLayerClient` with `Protect(..)` and `Release(..)` methods to protect and release multiple partitions at once.

> * Added additional checks for the result of cache related operations.

Reviewer: ```suggestion
* Added more checks for the results of cache-related operations.
```

> * Changed the `olp::dataservice::read::VersionedLayerClient::Release` to take into account the list of tiles from the request and assume that they will be released.

Reviewer: ```suggestion
* Updated `olp::dataservice::read::VersionedLayerClient::Release` to take the list of tiles from the request take into account and assume that they will be released.
```

> * Added performance improvements.

Reviewer: ```suggestion
* Introduced performance improvements.
```
